light_controlled_intersection_tactical_plugin/src/light_controlled_intersection_tactical_plugin.cpp

```diff
@@ -13,417 +13,377 @@
+                                                                        const GeneralTrajConfig &general_config, const DetailedTrajConfig &detailed_config)
     {
-      speed_i = std::max(config_.minimum_speed, algo_min_speed);
-      ROS_DEBUG_STREAM("Detected nan number from equations. Set to " << speed_i);
-    }
-
-    p.speed = std::max({speed_i, config_.minimum_speed, algo_min_speed});
-    p.speed = std::min({p.speed, speed_limit_, algo_max_speed}); 
-    ROS_DEBUG_STREAM("Applied speed: " << p.speed << ", at dist: " << total_dist_planned);
+        std::vector<PointSpeedPair> points_and_target_speeds;
+        
+        bool first = true;
+        std::unordered_set<lanelet::Id> visited_lanelets;
+        std::vector<carma_planning_msgs::msg::Maneuver> processed_maneuvers;
+        RCLCPP_DEBUG_STREAM(logger_->get_logger(), "VehDowntrack: "<<max_starting_downtrack);
+        for(const auto &maneuver : maneuvers)
+        {
+            double starting_downtrack = GET_MANEUVER_PROPERTY(maneuver, start_dist);
+            
+            starting_downtrack = std::min(starting_downtrack, max_starting_downtrack);
+
+            RCLCPP_DEBUG_STREAM(logger_->get_logger(), "Used downtrack: " << starting_downtrack);
+
+            // check if required parameter from strategic planner is present
+            if(GET_MANEUVER_PROPERTY(maneuver, parameters.float_valued_meta_data).empty())
+            {
+                throw std::invalid_argument("No time_to_schedule_entry is provided in float_valued_meta_data");
+            }
+
+            //overwrite maneuver type to use lane follow library function
+            carma_planning_msgs::msg::Maneuver temp_maneuver = maneuver;
+            temp_maneuver.type = carma_planning_msgs::msg::Maneuver::LANE_FOLLOWING;
+            RCLCPP_DEBUG_STREAM(logger_->get_logger(), "Creating Lane Follow Geometry");
+            std::vector<PointSpeedPair> lane_follow_points = basic_autonomy::waypoint_generation::create_lanefollow_geometry(maneuver, starting_downtrack, wm, general_config, detailed_config, visited_lanelets);
+            points_and_target_speeds.insert(points_and_target_speeds.end(), lane_follow_points.begin(), lane_follow_points.end());
+            
+            break; // expected to receive only one maneuver to plan
+        }
 
-    prev_point = p;
-    prev_speed = p.speed;
-  }
-}
+        //Add buffer ending to lane follow points at the end of maneuver(s) end dist 
+        if(!processed_maneuvers.empty() && processed_maneuvers.back().type == carma_planning_msgs::msg::Maneuver::LANE_FOLLOWING){
+            points_and_target_speeds = add_lanefollow_buffer(wm, points_and_target_speeds, processed_maneuvers, ending_state_before_buffer, detailed_config);
+        }
 
+        return points_and_target_speeds;
+    }
 
-} //namespace light_controlled_intersection_transit_plugin
\ No newline at end of file
+} // light_controlled_intersection_tactical_plugin
```

Comment:
The diff for this file is kind of a mess to read, I think git is a bit confused. But as far as i can read it it seems like things are in the right places.